sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala

```diff
@@ -98,6 +98,20 @@ class DataFrameSuite extends QueryTest with SharedSQLContext {
       testData.collect().toSeq)
   }
 
+  test ("union all") {
+    val unionDF = testData.unionAll(testData).unionAll(testData)
+      .unionAll(testData).unionAll(testData)
+
+    // Before optimizer, Union should be combined.
+    assert(unionDF.queryExecution.analyzed.collect {
+      case j @ Union(Seq(_, _, _, _, _)) => j }.size === 1)
```

Comment:
Sure, will change it. Actually, this is for checking if it has five children. We can explicitly check the number of children. Thanks!